### PR TITLE
Add Size::new_with_cross utility method

### LINKED ISSUE #547: Add Size::cross utility method
Not really related to this PR, but perhaps we can introduce something like `Size::cross` in a later PR instead of using this pattern :)

_Originally posted by @TimJentzsch in https://github.com/DioxusLabs/taffy/pull/545#discussion_r1343114148_

### PR DESCRIPTION
**Pull request by Taym95**
# Objective
Fixes #547

### CODE REVIEW

File: src/geometry.rs [hunk 1]
```diff
@@ -559,6 +559,17 @@ impl Size<Option<f32>> {
         Size { width: Some(width), height: Some(height) }
     }
 
+    /// Creates a new [`Size<Option<f32>>`] with either the width or height set based on the provided `direction`
+    pub fn new_with_cross(direction: FlexDirection, value: Option<f32>) -> Self {
```
Comment (Taym95) [inline src/geometry.rs:563]: can not be named `cross` since name is used in `Size<T>`

**Review by nicoburns — CHANGES_REQUESTED** [review]
PR generally looks good to me, but the name `new_with_cross` seems quite verbose.

I believe the method could be named just `cross` as static methods (no `self` parameter) do not conflict with object methods (with a `self` parameter). But perhaps the most idiomatic name would be `from_cross`. Could you change the name of the method to either `cross` or `from_cross`?

**Comment by alice-i-cecile** [comment]
I prefer `from_cross`

**Comment by nicoburns** [comment]
Checking the CI results, this method also needs to be feature flagged so that it only exists when the `flexbox` feature is enabled.

**Review by nicoburns — APPROVED** [review]

**Pull request merged**